### PR TITLE
feat: assignment algorithm

### LINKED ISSUE #24: Group assignment
we should implement some kind of algorithm to determine which group a newly created person is assigned to

### PR DESCRIPTION
**Pull request by KnutZuidema**
closes #24

### CODE REVIEW

File: pkg/api/group.go [hunk 1]
```diff
@@ -25,36 +25,44 @@ type GroupRepository interface {
...
+		groups:     groups,
+		groupAdder: adder,
 	}, nil
 }
 
 func (s GroupService) List(w http.ResponseWriter, r *http.Request) {
 	logger := s.logger.WithField("method", "list")
 	groups, err := s.groups.List(r.Context())
 	if err != nil {
+		logger.Error(err)
```
Comment (frzifus) [inline pkg/api/group.go:58]: i think we should return the error and use logging middleware to indicate errors.

**Review by frzifus — COMMENTED** [review]
lgtm, just one thing

File: pkg/api/group.go (hunk 1 above)
Comment (KnutZuidema) [reply pkg/api/group.go:58]: Do you mean this specific error or every one? Also returning the error means we have to wrap every function in some error handler or use some custom type for the function which implements `http.Handler`.

Comment (frzifus) [reply pkg/api/group.go:58]: That's right. All endpoints/handler that return an error then need this log middleware.

Comment (KnutZuidema) [reply pkg/api/group.go:58]: Alright sounds good to me, but that should be a separate pull request IMO.

Comment (frzifus) [reply pkg/api/group.go:58]: ok

**Review by frzifus — APPROVED** [review]
yolo :+1:

**Pull request merged**